src/xcode/ENA/ENA/Source/SceneDelegate.swift

```diff
@@ -118,15 +118,32 @@ final class SceneDelegate: UIResponder, UIWindowSceneDelegate, RequiresAppDepend
 
 	func userNotificationCenter(_: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
 		switch response.notification.request.identifier {
-		case ActionableNotificationIdentifier.testResult.identifier,
-			 ActionableNotificationIdentifier.riskDetection.identifier,
+		case ActionableNotificationIdentifier.riskDetection.identifier,
 			 ActionableNotificationIdentifier.deviceTimeCheck.identifier:
 			showHome(animated: true)
 
 		case ActionableNotificationIdentifier.warnOthersReminder1.identifier,
 			 ActionableNotificationIdentifier.warnOthersReminder2.identifier:
-			showPositiveTestResultFromNotification(animated: true)
+			checkIfPositiveTestResultWasShown()
 
+		case ActionableNotificationIdentifier.testResult.identifier:
+			guard let testResultRawValue = response.notification.request.content.userInfo[ActionableNotificationIdentifier.testResult.identifier] as? Int,
+				  let testResult = TestResult(rawValue: testResultRawValue) else {
+				showHome(animated: true)
+				return
+			}
+			
+			switch testResult {
+			case .positive, .negative:
+				showTestResultFromNotification(with: testResult, animated: true)
+				
+			case .invalid:
+				showHome(animated: true)
+				
+			default:
```

Comment:
Did you consider, to explicitly handling all cases? This way, if a new case is added, the compiler will fail here and the future-developer has to think about showing some UI or not for the new case.

Reply:
so make a fatal error for the default case? I thoguht about it actually but i didn't see it alot in the code here. ok will do it